packages/typescript/src/index.ts

```diff
@@ -35,7 +35,7 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
   const watchProgramHelper = new WatchProgramHelper();
 
   const parsedOptions = parseTypescriptConfig(ts, tsconfig, compilerOptions);
-  const filter = createFilter(include || ['*.ts+(|x)', '**/*.ts+(|x)'], exclude, {
+  const filter = createFilter(include || ['*.(|c|m)ts+(|x)', '**/*.(|c|m)ts+(|x)'], exclude, {
```

Comment:
I must admit I did not test this against micromatch/createFilter, but at least Bash with extended globbing (where this syntax comes from) does not recognize this as a valid pattern because `(` needs to be preceded by a special symbol, see e.g. https://www.linuxjournal.com/content/bash-extended-globbing
So I would guess the pattern should be more like

```
**/*.?(c|m)ts?(x)
```

Also I wonder if we just need the second version starting with `**`. But again, I did not test this.

Reply:
Ah, thanks for catching these! Yeah the glob there was off, failed to note the `+` in front of `(|x)` when adapting.
I tried a new one with a little trick I picked up way back:

```
Welcome to Node.js v16.15.1.
Type ".help" for more information.
> const glob = require('glob')
undefined
> glob.sync('{,**/}*.?(c|m)ts?(x)')
[
  'node_modules/@cspotcode/source-map-support/register-hook-require.d.ts',
  'node_modules/@cspotcode/source-map-support/register.d.ts',
  'node_modules/@cspotcode/source-map-support/source-map-support.d.ts',
  'node_modules/@types/babel__core/index.d.ts',
  'node_modules/@types/babel__generator/index.d.ts',
  'node_modules/@types/babel__template/index.d.ts',
  ... 515 more items
]
```

It compresses both globs into one, seems to work!